### DISCUSSION TITLE
Style Guide

### DISCUSSION THREAD

**Discussion by troyhunt** [HaveIBeenPwned/ux-rebuild#8] [Ideas]
More to come, jotting this down during weekly video.

**Comment by neilcaldwell** [comment]
A web design style guide, also known as a digital style guide, is a comprehensive resource that consolidates and presents all the design decisions for a website. It serves as a reference for maintaining consistency and cohesion in the visual language used across the site. Let’s delve into the details:

### What Is a Web Design Style Guide?
        A style guide encompasses various design elements, including:
            Color scheme: Defining the colors to be used consistently.
            Typography: Establishing rules for fonts, sizes, and styles.
            Spacing and layout: Setting guidelines for margins, padding, and overall page structure.
            Icons and imagery: Describing how icons and images should be used.
            Visual language: Collecting all visual elements that contribute to the website’s aesthetics.
        It acts as a go-to resource for maintaining brand consistency and ensuring that everyone involved in the project adheres to the same visual language.

### Why Should You Create a Style Guide?
        Efficiency: Having a style guide streamlines the design process and reduces ambiguity.
        Consistency: It ensures that your website maintains a uniform look and feel across different pages and sections.
        Brand identity: A style guide reinforces your brand’s visual identity and voice.

###     How to Create a Web Design Style Guide:
        Step 1: Study Your Brand:
            Understand your brand’s core values, mission, and target audience.
        Step 2: Determine Your Logo Usage Rules:
            Specify how and where your logo should appear.
        Step 3: Define Your Color Palette:
            Choose a consistent set of colors for various elements.
        Step 4: Create Rules for Typography:
            Decide on font families, sizes, and styles.
        Step 5: Set Layout and Spacing Rules:
            Establish guidelines for margins, padding, and overall layout.
        Step 6: Consider Icon Style:
            Define the style and usage of icons.
        Step 7: Define Guidelines for Illustrations and Imagery:
            Specify how illustrations and images should be used.
        Step 8: Outline Stylistic Considerations:
            Address any other stylistic aspects relevant to your brand.
        Step 9: Develop Your Brand’s Voice:
            Describe the tone and language to be used in content.
        Step 10: Include Specific Dos and Don’ts:
            Provide clear instructions on what to do and what to avoid.
        Step 11: Don’t Forget About Downloadable Assets:
            [Include downloadable versions of logos, icons, and other assets](https://elementor.com/blog/web-design-style-guide/)

If you’re looking for a starting point, consider using our Style Guide Template, which includes examples for logos, fonts, colors, buttons, and icons. [It’ll help you create your own style guide efficiently! 🎨📝 ](https://elementor.com/blog/web-design-style-guide/)

**Comment by neilcaldwell** [comment]
Colours are easy changed, and also, hard to pick. Starting with the colours of the old site, I'd choose something similar, using: [ui colours](https://uicolors.app/create) a blue looks nice to start with. 

![image](https://github.com/HaveIBeenPwned/ux-rebuild/assets/12889992/2113d5a1-61dc-4548-93ca-9c0e42913b2a)

**Comment by markotsoi** [comment]
One thing to consider for the styleguide (and website colours in general), is that the current colour scheme doesn't meet WCAG AA or AAA accessibility guidelines, for instance, the text below the email address input field on the homepage is too light: https://webaim.org/resources/contrastchecker/?fcolor=5B9FBD&bcolor=2E82A7.

**Comment by the-honey-badger-dev** [comment]
From a practical standpoint, an interesting implementation of Styleguide in one of my previous jobs was to use Styleguide artefacts themselves as the source of styling in applications, something I suggested early on.
This was for a company which owned several client facing brands (12 of them) and over time we managed to white-label all of our applications.
Later on, I worked on the build, versioning and publishing system for the Styleguide team which allowed application teams to upgrade their look and feel over time.
The main takeaways were a) use Styleguide itself as the source of styling for the applications and b) manage styling changes as versions.
